react/src/views/Box/BoxView.test.tsx

```diff
@@ -0,0 +1,416 @@
+        let numberOfItemWhenIncreased = 32;
+        // let numberOfItemWhenDecreased = 31;
+        fireEvent.click(screen.getByTestId("increase-items"));
+        await waitFor(() => userEvent.type(screen.getByTestId("increase-number-items"), "1"));
+        fireEvent.click(screen.getByText("Submit"));
+
+        await waitFor(() => {
+            expect(screen.getByTestId("boxview-number-items")).toBeInTheDocument();
+            expect(screen.getByTestId("boxview-number-items")).toHaveTextContent(
+                "# " + numberOfItemWhenIncreased,
+            );
+        });
+
+        // fireEvent.click(screen.getByTestId("decrease-items"));
+        // await waitFor(() => userEvent.type(screen.getByTestId("decrease-number-items"), "1"));
+
+        // await waitFor(() => {
+        //     expect(screen.getByTestId("boxview-number-items")).toBeInTheDocument();
+        //     expect(screen.getByTestId("boxview-number-items")).toHaveTextContent(
+        //         "# " + numberOfItemWhenDecreased,
+        //     );
+        // });
+    });
+    // Test case 3.1.1.4
+    it("3.1.1.4 - clicking on Lost / Scrap must change box state respectfully", async () => {
+        await waitFor(waitTillLoadingIsDone);
+
+        fireEvent.click(screen.getByTestId("box-lost-btn"));
+        await waitFor(() => {
+            const boxSubheading = screen.getByTestId("box-subheader");
+            expect(boxSubheading).toHaveTextContent(`State: Lost`);
+            expect(screen.getByTestId("box-state")).toHaveStyle(`color: #EB404A`);
+        });
+
+        fireEvent.click(screen.getByTestId("box-scrap-btn"));
+        await waitFor(() => {
+            const boxSubheading = screen.getByTestId("box-subheader");
+            expect(boxSubheading).toHaveTextContent(`State: Scrap`);
+            expect(screen.getByTestId("box-state")).toHaveStyle(`color: #EB404A`);
+        });
+    });
+    // Test case 3.1.1.5
+    it("3.1.1.5 - click on a location should move box to seleted place", async () => {
+        await waitFor(waitTillLoadingIsDone);
+
+        // fireEvent.click(screen.getByTestId("location-wh1-btn"));
+        // await waitFor(() => {
+        //     const boxLocationLabel = screen.getByTestId("box-location-label");
+        //     expect(boxLocationLabel).toHaveTextContent(`Move this box from WH1 to:`);
+        // });
+    });
+
+    // Test case 3.1.1.6
+    it("3.1.1.6 - If Distro Event Not Available it should not shown the ui section for that", async () => {
+        await waitFor(waitTillLoadingIsDone);
+        const distroEventSection = screen.getByTestId("distro-event-section");
+        expect(distroEventSection).toBeInTheDocument();
+    });
+    // Test case 3.1.1.7
+    it("3.1.1.7 - render tags correctly if box assigned tags", async () => {
+        await waitFor(waitTillLoadingIsDone);
+        const boxTags = screen.getByTestId("box-tags");
+        expect(boxTags).toBeInTheDocument();
```

Comment:
I noticed there are `tags` data in the mocks, is it hard to verify that the tag is actually shown in the UI?

Reply:
@pylipp I will try to verify `tags` per your suggestion